rcl/include/rcl/wait.h

```diff
@@ -0,0 +1,306 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__WAIT_H_
+#define RCL__WAIT_H_
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdbool.h>
+#include <stddef.h>
+
+#include "rcl/subscription.h"
+#include "rcl/guard_condition.h"
+#include "rcl/time.h"
+#include "rcl/types.h"
+
+struct rcl_wait_set_impl_t;
+
+/// Container for subscription's, guard condition's, etc to be waited on.
+typedef struct rcl_wait_set_t {
```

Comment:
That sounds fine with me. It should be an immutable value once specified,
so the method you described works as the way to enforce it.

On Thu, Dec 3, 2015 at 1:57 PM, William Woodall <notifications@github.com>
wrote:

> In rcl/include/rcl/wait.h
> <https://github.com/ros2/rcl/pull/5#discussion_r46618807>:
>
> > +extern "C"
> > +{
> > +#endif
> > +
> > +#include <stdbool.h>
> > +#include <stddef.h>
> > +
> > +#include "rcl/subscription.h"
> > +#include "rcl/guard_condition.h"
> > +#include "rcl/time.h"
> > +#include "rcl/types.h"
> > +
> > +struct rcl_wait_set_impl_t;
> > +
> > +/// Container for subscription's, guard condition's, etc to be waited on.
> > +typedef struct rcl_wait_set_t {
>
> Maybe, though I think it would be passed as an option to the
> rcl_wait_set_init function and the result would be hidden behind an
> accessor and stored in the impl pointer.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ros2/rcl/pull/5/files#r46618807>.
>